server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/statements/SetUpDownByStatement.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2021 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.core.model.tree.statements;
+
+import com.google.common.collect.ImmutableList;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+import lombok.Value;
+import org.eclipse.lsp.cobol.core.model.ErrorSeverity;
+import org.eclipse.lsp.cobol.core.model.Locality;
+import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp.cobol.core.model.variables.ElementItem;
+import org.eclipse.lsp.cobol.core.model.variables.IndexItem;
+import org.eclipse.lsp.cobol.core.model.variables.Variable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** This class implements the logic for SET UP/DOWN BY statement. */
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class SetUpDownByStatement extends StatementNode {
+  private static final List<String> RECEIVING_FIELD_TYPES = ImmutableList.of("variables.indexName");
+  private static final List<String> SENDING_FIELD_TYPES =
+      ImmutableList.of("variables.elementaryInteger", "variables.nonzeroInteger");
+  List<Locality> receivingFields;
+  Locality sendingField;
+  String literal;
+
+  public SetUpDownByStatement(
+      Locality locality, List<Locality> receivingFields, Locality sendingField, String literal) {
+    super(locality);
+    this.receivingFields = receivingFields;
+    this.sendingField = sendingField;
+    this.literal = literal;
+  }
+
+  @NonNull
+  @Override
+  public List<SyntaxError> validate(Map<Locality, Variable> variableUsages) {
+    List<SyntaxError> errors = new ArrayList<>();
+    receivingFields.forEach(
+        it -> {
+          Variable variable = variableUsages.get(it);
+          if (variable != null && !(variable instanceof IndexItem)) {
+            errors.add(
+                createSyntaxError(it, "statements.invalidReceivingField", RECEIVING_FIELD_TYPES));
+          }
+        });
+    if (literal != null) {
+      if (!literal.matches("[+-]?\\d+") || Integer.parseInt(literal) == 0) {
```

Comment:
If assigned to meaningful named field: String signedNumPattern="[+-]?\\d+"   then no need to merge two IFs